scapy/modules/packet_viewer/show_view.py

```diff
@@ -0,0 +1,227 @@
+from scapy.packet import Packet
+from scapy.themes import BlackAndWhite
+from scapy.utils import hexdump
+from scapy.modules.packet_viewer.extended_listbox import ExtendedListBox
+from scapy.modules.packet_viewer.details_view import DetailsView
+from scapy.modules.packet_viewer.field_edit import FieldEdit
+
+
+class ShowView(DetailsView):
+    """
+    Custom view which holds the output of Packet.show() as editable list and
+    shows a hexdump of the current selected Packet
+    """
+    action_name = "Edit"
+
+    def __init__(self):
+        # type: () -> None
+        """
+        Initialize ShowView
+        """
+        self._current_packet = None
+        self._show_text = ExtendedListBox(False, SimpleListWalker([]))
+        self._hex_text = ExtendedListBox(False, SimpleListWalker([]))
+
+        hexdump_str_width = 71
+        col = Columns([self._show_text, (hexdump_str_width, self._hex_text)],
+                      dividechars=2)
+        super(ShowView, self).__init__(col)
+
+    def update_packets(self, focused_packet, all_packets):
+        # type: (Packet, List[Packet]) -> None
+        self._update(focused_packet)
+
+    def _update(self, packet, force_update=False):
+        # type: (Packet, Optional[bool]) -> None
+        """
+        Internal update function
+        :param packet: Packet which get displayed by this view
+        :param force_update: Forces re-rendering
+        """
+        if packet == self._current_packet and not force_update:
+            return
+
+        self._current_packet = packet
+
+        show_text = self._show(packet) + [Text("")]
+        hexdump_text = hexdump(packet, dump=True)
+
+        # Keep an empty line as the last line. This gives a nice
+        # visual feedback that the end of the list is reached.
+        # For `show_text` this is given because it always ends with an "\n"
+        # For `hexdump_text` we add it manually
+        self._update_hexdump(hexdump_text.split("\n"))
+        self._update_show(show_text)
+
+    @staticmethod
+    def _build_command(target_type, string):
+        # type: (Type, str) -> Any
+        """
+        This method tries to build a value from a string for any type.
+        :param target_type: desired type of string
+        :param string: string that should be build to a value
+        :return: value
+        """
+        try:
+            # For Python3 we need to add the "b" prefix for bytes
+            # Python2 does not need this
+            if target_type == bytes and six.PY3:
+                value = literal_eval(
+                    'b"' + string[1:-1].replace('"', '\\"') + '"')
+            else:
+                value = literal_eval(string)
+        except (SyntaxError, ValueError):
+            # Encapsulate value_str as a string and parse as simple string
+            # Should always work except if the field doesn't accept a string
+            value = literal_eval('"' + string.replace('"', '\\"') + '"')
```

Comment:
Sollte das überprüft werden, ob das Feld keinen string akzeptiert?

Reply:
@akorb made this.